File: Second-semester/Java/Competitive/Irreducible fractions/Fractions.java
import java.util.Scanner;

public class Fractions {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int i, input, counter = 0;

        do{ 
            input = Integer.parseInt(sc.nextLine());

            for(i = 1; i < input; i++){
                if(gcd(i, input) == 1){
                    counter++;
                }
            }
            System.out.println(counter);
            counter = 0;
        }while(input != 0);
        
    }
    public static int gcd(int n, int m){
        if(m==0){
            return n;
        }else{
            return gcd(m, n%m);
        }
    }
}

File: Second-semester/C++/Irreducible Fractions/fractions.cpp
#include <iostream>
using namespace std;

int gcd(int n, int m){
    if(m==0){
        return n;
    }else{
        return gcd(m, n%m);
    }
}

int main(){
    int i, input, counter = 1;

    do {
        cin >> input;

        for (i = 1; i <= input; i++)
        {
            if (gcd(i, input) == 1) {
                counter++;
            }
        }
        cout << counter;
        counter = 0;
    } while (input != 0);

    return 0;
}